### DISCUSSION TITLE
A question about Spawn of Energy usage

### DISCUSSION THREAD

**Discussion by shiyishiba** [lbl-srg/modelica-buildings#3828] [Q&A]
I am trying to use the concept of Spawn of EnergyPlus in the buildings library. Under Buildings/ThermalZone/EnergyPlus9_6_0 this path have a called AirHeating case, this case have a module that made me feel confused, this module is called building.
The module is described as: [Model that declares building-level specifications for Spawn of EnergyPlus.
This model is used to configure EnergyPlus. Each EnergyPlus idf file must have one instance of this model,  and the instance name must be building. The instance must be placed in the model hierarchy at the same or at a higher  level than the EnergyPlus objects that are related to the EnergyPlus idf file specified in this model through the  parameter idfName.] I can upload an idf file, an epw file, and a mos file in this module. This module, in the case of AirHeating, appears to provide outdoor environment parameters. I took a screenshot of the problem I encountered and attached it below. Figure 1 is the expression of the case of AirHeating. The red boxes are where I am confused. What is their function? Figure 2 shows the details after clicking Building in detail. Ask a professional to clear my mind. By the way, if I want to simulate energy consumption, for example, what is the heat flow in a room on a given day, which modules should I use? Thank you for reading this.
![problem3](https://github.com/lbl-srg/modelica-buildings/assets/125559199/ff73c49d-152c-4b77-ae7c-c61a59aef80e)

![problem2](https://github.com/lbl-srg/modelica-buildings/assets/125559199/d65c8018-d63a-42d8-a478-f56e34e3bae1)

**Comment by hcasperfu** [comment]
Greetings!
I am not entirely clear what your main question is. I will answer to the best of my understanding.

The `pAtm` component sets the boundary condition of the air loop. You can think of it as the pressurisation point of the air loop. For example, for a closed fluid loop with components $A$, $B$, and $C$ whose pressure relations can be described as $p_A=p_B$, $p_B=p_C$, and $p_C=p_A$, this system has no solution until a boundary condition such as $p_A=5$ is specified. `pAtm` provides such a condition.
The component `building` configures EnergyPlus. It has no graphical representation except for the weather instance and the bus connector that you see in your first screenshot. You can see more content in the text mode. Otherwise, this component provides weather conditions to the model through the bus connector so that the model knows the properties of the outside air.
The amount of heat provided to the room through the heater can be visited via the variable `hea.Q_flow`.

Cheers,
Casper
